libvirtrpc: flappy test: TestEvents

```
=== RUN   TestEvents
--- FAIL: TestEvents (1.13s)
	rpc_test.go:472: expected event, received timeout
	rpc_test.go:477: expected device "drive-ide0-0-0", got %!q(<nil>)
```

Can be reproduced in a shell:
```
while true; do go test -v -run=TestEvents github.com/digitalocean/go-qemu/qmp/libvirtrpc ; done
```

--- a/qmp/libvirtrpc/rpc_test.go
+++ b/qmp/libvirtrpc/rpc_test.go
@@ -460,20 +460,22 @@ func TestEvents(t *testing.T) {
 		t.Error(err)
 	}
 
-	// send an event
 	go func() {
-		conn.test.Write(append(testEventHeader, testEvent...))
+		var e qmp.Event
+		select {
+		case e = <-stream:
+		case <-time.After(time.Second * 1):
+			t.Error("expected event, received timeout")
+		}
+
+		expected := "drive-ide0-0-0"
+		if e.Data["device"] != expected {
+			t.Errorf("expected device %q, got %q", expected, e.Data["device"])
+		}
 	}()
 
-	var e qmp.Event
-	select {
-	case e = <-stream:
-	case <-time.After(time.Second * 1):
-		t.Error("expected event, received timeout")
-	}
-
-	expected := "drive-ide0-0-0"
-	if e.Data["device"] != expected {
-		t.Errorf("expected device %q, got %q", expected, e.Data["device"])
-	}
+	// send an event to the listener goroutine; since this will
+	// block, the listener should get scheduled to receive the
+	// value immediately
+	conn.test.Write(append(testEventHeader, testEvent...))
 }

